idaes/core/util/model_diagnostics.py

```diff
@@ -251,6 +267,14 @@ def svd_sparse(jacobian, number_singular_values):
         description="Tolerance for raising a warning for small Jacobian values.",
     ),
 )
+CONFIG.declare(
+    "warn_for_evaluation_error_at_bounds",
```

Comment:
@Robbybp, @andrewlee94 - what do you think of this name?

Reply:
Sounds good to me.

Reply:
Looks good to me

Reply:
Awesome. Thank you both.